File: LeetCode/Kotlin/s2486.kt
class Solution2486 {
    fun appendCharacters(s: String, t: String): Int {
        var j = 0
        var i = 0
        while (i < s.length && j < t.length) if (s[i++] == t[j]) j++
        return t.length - j
    }
}

File: LeetCode/Rust/s2486.rs
struct Solution2486 {}

impl Solution2486 {
    pub fn append_characters(s: String, t: String) -> i32 {
        let mut j:usize = 0;
        let mut i:usize = 0;
        let s = s.as_bytes();
        let t = t.as_bytes();
        while i < s.len() && j < t.len() { if s[i] == t[j] {j+=1}; i +=1 }
        return (t.len() - j) as i32
    }
}